### PR TITLE
RELATED: TNT-1107 Attribute filter formatting

### PR DESCRIPTION
**Pull request by marmil-cz**
---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10314/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12302/ : SUCCESS in 23m 47s

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts [hunk 1]
```diff
@@ -130,6 +140,13 @@ class TigerWorkspaceElementsQuery implements IElementsQuery {
         return {};
     }
 
+    private getGranularity(identifier: string) {
```
Comment (scavnickyj) [inline libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts:143]: This is no longer needed as we get the granularity from the backend response itself.

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts [hunk 2]
```diff
@@ -167,15 +184,33 @@ class TigerWorkspaceElementsQuery implements IElementsQuery {
...
                 });
 
-                const { paging, elements } = response.data;
+                const { paging, elements, format } = response.data;
+
+                const refAsValue = objRefToString(this.ref);
+                const granularity = this.getGranularity(refAsValue);
+                const sdkGranularity = toSdkGranularity(granularity);
+                const locale = format?.locale as any;
+                const pattern = format?.pattern as any;
```
Comment (scavnickyj) [inline libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts:193]: Can we rather use type cast to `FormattingLocale` from defaultDateFormatter? And pattern is always a string.

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts [hunk 3]
```diff
@@ -167,15 +184,33 @@ class TigerWorkspaceElementsQuery implements IElementsQuery {
...
+                const dateValueFormatter = createDateValueFormatter(this.dateFormatter);
 
                 return {
-                    items: elements.map(
-                        (element): IAttributeElement => ({
+                    items: elements.map((element): IAttributeElement => {
+                        const objWithFormattedTitle = sdkGranularity
```
Comment (scavnickyj) [inline libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts:198]: Condition `format && sdkGranularity` would be safer here.

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts (hunk 1 above)
Comment (marmil-cz) [reply libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts:143]: ok

**Comment by marmil-cz** [comment]
extended test

**Comment by marmil-cz** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|17.8700s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|20.7460s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|15.1310s|
|✅|**barChartDrilling.spec.ts**|4|-|7.2660s|
|✅|**pivotTableSizing.spec.ts**|1|-|5.0520s|
|✅|**headlineChartDrilling.spec.ts**|2|-|3.9250s|
|✅|**dateFilter.spec.ts**|9|-|15.0960s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|7.5400s|
|✅|**dashboard.spec.ts**|9|-|53.4160s|
|✅|**attributeFilterButton.spec.ts**|9|-|31.2780s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|39.8610s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|7.3170s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|7.3250s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|7.9840s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/322/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10325/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/322/ : SUCCESS in 13m 35s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10323/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12310/ : SUCCESS in 21m 59s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1882/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10324/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1882/ : SUCCESS in 25m 59s

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts (hunk 2 above)
Comment (marmil-cz) [reply libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts:184]: Changed

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts (hunk 3 above)
Comment (marmil-cz) [reply libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts]: added

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10327/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12313/ : FAILURE in 8m 33s

File: libs/sdk-ui-ext/src/internal/utils/colors.ts [hunk 4]
```diff
@@ -18,11 +18,11 @@ import {
     isMeasureDescriptor,
     isResultAttributeHeader,
 } from "@gooddata/sdk-model";
-import { getMappingHeaderName, IColorAssignment, IMappingHeader } from "@gooddata/sdk-ui";
+import { getMappingHeaderFormattedName, IColorAssignment, IMappingHeader } from "@gooddata/sdk-ui";
 import { ColorUtils } from "@gooddata/sdk-ui-charts";
 
 function getItemName(item: IColoredItem): string {
-    return getMappingHeaderName(item.mappingHeader) || "";
+    return getMappingHeaderFormattedName(item.mappingHeader) || "";
```
Comment (scavnickyj) [inline libs/sdk-ui-ext/src/internal/utils/colors.ts:25]: This is not needed anymore and was part of my changes for execution. Looks like some leftovers after rebase from the past.

File: libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts [hunk 5]
```diff
@@ -167,15 +177,33 @@ class TigerWorkspaceElementsQuery implements IElementsQuery {
...
+                const locale = format?.locale as FormattingLocale;
+                const pattern = format?.pattern as string;
+                const dateValueFormatter = createDateValueFormatter(this.dateFormatter);
 
                 return {
-                    items: elements.map(
-                        (element): IAttributeElement => ({
+                    items: elements.map((element): IAttributeElement => {
+                        const objWithFormattedTitle =
+                            sdkGranularity && format
```
Comment (scavnickyj) [inline libs/sdk-backend-tiger/src/backend/workspace/attributes/elements/index.ts:191]: Maybe extract this to a property above the map to something like `shouldFormatTitle`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10328/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12314/ : SUCCESS in 20m 25s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10329/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12315/ : SUCCESS in 24m 02s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3143

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/24/ : FAILURE in 15m 47s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3290/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2138/ : SUCCESS in 17m 02s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12318/ : SUCCESS in 21m 11s

**Pull request merged**